internal/credentials/gar/workload_identity_federation.go

```diff
@@ -0,0 +1,135 @@
+package gar
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/patrickmn/go-cache"
+	"google.golang.org/api/iamcredentials/v1"
+
+	"github.com/akuity/kargo/internal/logging"
+)
+
+var garURLRegex = regexp.MustCompile(`^.+\.pkg\.dev/`)
```

Comment:
Though google has deprecated Google Container Registry, the `gcr.io` domain is still valid and [is possible to be hosted by artifact registry](https://cloud.google.com/artifact-registry/docs/transition/gcr-on-ar) by existing gcr.io users.

I think this should be relaxed to also check for `gcr.io`

Reply:
Nice catch. Will relax.

Reply:
Fixed in https://github.com/akuity/kargo/pull/2070/commits/f648029daf0f199c0f193328fac3ccaf27eaee92